components/suggestion/command_provider/app_command_parser.ts

```diff
@@ -43,72 +41,582 @@ export type Store = {
+                    i++;
+                }
+                parsed.state = ParseState.Flag;
+                parsed.incomplete = '';
+                parsed.incompleteStart = i;
+                flagEqualsUsed = false;
+                break;
+            }
+
+            case ParseState.Flag: {
+                switch (c) {
+                case '': {
+                    if (autocompleteMode) {
+                        return parsed;
+                    }
+
+                    // for submit fall through to whitespace, to handle an (implicit) BOOl value.
+                }
+                // eslint-disable-next-line no-fallthrough
+                case ' ':
+                case '\t':
+                case '=': {
+                    const field = fields.find((f) => f.label === parsed.incomplete);
+                    if (!field) {
+                        return this.parseError('command does not accept flag ' + parsed.incomplete + ': ' + i);
+                    }
+                    parsed.state = ParseState.FlagValueSeparator;
+                    parsed.field = field;
+                    parsed.incomplete = '';
+                    break;
+                }
+                default: {
+                    parsed.incomplete += c;
+                    i++;
+                    break;
+                }
+                }
+                break;
+            }
+
+            case ParseState.FlagValueSeparator: {
+                parsed.incompleteStart = i;
+                switch (c) {
+                case '': {
+                    if (autocompleteMode) {
+                        return parsed;
+                    }
+                    parsed.state = ParseState.StartValue;
+                    break;
+                }
+                case ' ':
+                case '\t': {
+                    i++;
+                    break;
+                }
+                case '=': {
+                    if (flagEqualsUsed) {
+                        return this.parseError('multiple = signs are not allowed: ' + i);
+                    }
+                    flagEqualsUsed = true;
+                    i++;
+                    break;
+                }
+                default: {
+                    parsed.state = ParseState.StartValue;
+                }
+                }
+                break;
+            }
+
+            case ParseState.StartValue: {
+                parsed.incomplete = '';
+                parsed.incompleteStart = i;
+                switch (c) {
+                case '"': {
+                    parsed.state = ParseState.QuotedValue;
+                    i++;
+                    break;
+                }
+                case '`': {
+                    parsed.state = ParseState.TickValue;
+                    i++;
+                    break;
+                }
+                case ' ':
+                case '\t':
+                    return this.parseError('unreachable: unexpected whitespace: ' + i);
```

Comment:
not sure if it is a goal for test cases to exercise all `parseErrors` or not.

Reply:
It was not the goal yet, but the unreachables are not testable anyway. They are just there as precautions.